src/views/EditUserView.vue

```diff
@@ -0,0 +1,240 @@
+      <br>
+    </div>
+
+    <div class="row">
+      <div class="col-md-6">
+        <label>Phone Number:</label>
+        <input
+          v-model="phoneNumber"
+          type="text"
+          name="phoneNumber"
+          class="form-control"
+          required
+          placeholder="Phone Number"
+        >
+      </div>
+
+      <div class="col-md-6">
+        <label>Institutional Affiliation:</label>
+        <input
+          v-model="affiliation"
+          type="text"
+          name="affiliation"
+          class="form-control"
+          required
+          placeholder="Institutional Affiliation"
+        >
+        <div class="invalid-feedback">
+          Please fill out this field.
+        </div>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <br>
+      <button
+        type="success"
+        name="submit"
+        class="btn btn-primary"
+        style="margin: auto"
+        @click="updateUser"
+      >
+        Register
+      </button>
+    </div>
+    <div
+      v-if="regError"
+      class="alert alert-danger"
+      role="alert"
+      name="regError"
+    >
+      {{ regError }}
+    </div>
+  </form>
+</template>
+
+<script setup>
+  import{ ref, watch } from 'vue';
+  import { useRouter } from 'vue-router';
+  import { firebaseStore, collection, query, where, getDocs, updateDoc, doc } from '../firebase/database';
+  import { getAuth } from '../firebase/firebaseAuth';
+  import { useUserStore } from '../stores/user';
+  const firstName = ref('');
+  const surname = ref('');
+  const address1 = ref('');
+  const address2 = ref('');
+  const address3 = ref('');
+  const city = ref('');
+  const postCode = ref('');
+  const phoneNumber = ref('');
+  const affiliation = ref('');
+  const router = useRouter();
+  const regError = ref('');
+  const auth = getAuth();
+
+
+  const userStore = useUserStore();
+  const user = userStore.user;
+  firstName.value = user.info.firstName;
+  surname.value = user.info.surname;
+  address1.value = user.info.addressline1;
+  address2.value = user.info.addressline2;
+  address3.value = user.info.addressline3;
+  city.value = user.info.city;
+  postCode.value = user.info.postCode;
+  phoneNumber.value = user.info.phoneNumber;
+  affiliation.value = user.info.affiliation;
+
+  watch(phoneNumber, () =>{
+    if(/(^\+?[0-9]{10,15})$/.test(phoneNumber.value) ){
+      regError.value = null;
+    }else{
+      regError.value = 'Phone number is invalid';
+    }
+  });
+
+  async function updateUser(){
+    if(!regError.value){
+      if(!firstName.value || !surname.value || !address1.value|| !city.value || !postCode.value ||!phoneNumber.value
+        || !affiliation.value){
+        regError.value='Please fill all fields';
+      }else{
+        const getUserId = query(collection(firebaseStore,'users_dev'), where('uid','==',auth.currentUser.uid));
+        const getData = await getDocs(getUserId);
+        console.log(getData);
+        getData.forEach(async (docData)=>{
```

Comment:
No longer need to manually find the user if you are using the user store.

Reply:
removed catch and the console log inside of it
![image](https://user-images.githubusercontent.com/46050460/233805979-f477730b-0554-448e-858d-bdd0e5ba1dfc.png)

Reply:
I think i still need to query it since the document ID is different to the user ID example below:
![image](https://user-images.githubusercontent.com/46050460/233806911-f5c72fcd-75d7-4603-b76d-42ccf7381ef0.png)

Reply:
But do you still need the getUserId? if you are passing that into getdocs?

Reply:
Do we need that for each though? it is implying that a user could have multiple documents which shouldn't be the case? If the requirement is to get the doc id for update doc, would it be better to store the doc id in the user info object in the store?

Reply:
getUserId is named incorrectly it should be called getDocId which i will change, but it is required since the query searches through the user database, then gets the user UID and finally queries through the user database to get the document. Then the getDocs just gets the documents where the id is found and the updateDoc code just updates the documents selected.

The reason im using getDocs is that when querying for a document ID using getDoc does not work since the query returns a object inwhich getDoc doesnt accept but getDocs does. Also ive changed the code so that it only retrieves 1 document as shown below:
![image](https://user-images.githubusercontent.com/46050460/233808776-2c810f31-de79-4a86-9873-578cd5d48a87.png)
It currently works with this code

Reply:
Like I said, could we not store the user doc id in the pinia store rather than querying the user collection to get the document id?

Reply:
So instead of doing getDocId and getData, we can just do user.info.docID for update doc function?

Reply:
If you are updating the doc, you also have to update the pinia store user.info as well. Not sure if you update a user profile and go back to the profile page, does the changes reflect what is on there?

Reply:
yeah we could store it when the user registers I think, but im not sure how to get the document id and store it in the pinia.
And when updating the pinia would i have to do something like:
```
user.info.firstName: firstName.value
``` 
?

Reply:
You do not store it on register. The user store is populated when the user logs in successfully. In fact, we already have the query snap shot. Look at stores/user.js. it currently stores the .data() specifically. We can keep user.info intact since it is already being used on the profile page, instead, you can make a new user store property called 'doc' which stores the query snap shot. Then, you will have access to the doc id without much additional work.

Reply:
Yes
```
user.info.firstName: firstName.value
```

Reply:
Made changes to user.js and edituserview.vue to allow it to get data from the pinia instead of calling a query:
![image](https://user-images.githubusercontent.com/46050460/233811900-5b17d233-3bcc-4810-abc7-a048bf919cc5.png)
I added a line which retrieves the doc id and inserted a new store named doc which contains the document id 
Inside of EditUser vue just retrieved the doc and placed it inside of updateDoc
![image](https://user-images.githubusercontent.com/46050460/233811950-6703a64c-e1ac-4f73-8496-e4ab5377de05.png)

Reply:
Added assigned data to pinia aswell once the authentication is complete 
![image](https://user-images.githubusercontent.com/46050460/233811959-391f537c-4178-4411-8037-d5c434f7ebe1.png)

Reply:
Refer to code comments. Resolving this comment chain as the discussion will take place in the other code comments. The pinia store assignment looks fine. Just remember to test that it works.